brave/src/main/java/brave/RealSpan.java

```diff
@@ -140,13 +140,12 @@
   }
 
   @Override public void finish() {
-    finish(clock.currentTimeMicroseconds());
+    finish(0L);
```

Comment:
@adriancole would you mind sharing the reasoning behind `0L` instead of `#currentTimeMicroseconds()`

Reply:
to prevent overhead of using the clock when the span is already finished.

Reply:
TODO: comment

Reply:
fixed upstream